plugins/processors/t128_transform/t128_transform.go

```diff
@@ -250,6 +279,36 @@ func (r *T128Transform) Init() error {
 	return nil
 }
 
+func loadCache(path string) (map[uint64]map[string]observedValue, error) {
+	cache := make(map[uint64]map[string]observedValue)
+
+	data, err := ioutil.ReadFile(path)
+	if err != nil {
+		return cache, err
+	}
+
+	err = json.Unmarshal(data, &cache)
+	if err != nil {
+		return cache, err
+	}
+
+	return cache, nil
+}
+
+func persistCache(path string, cache map[uint64]map[string]observedValue) error {
+	file, err := json.MarshalIndent(cache, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(path, file, 0644)
```

Comment:
does the directory need to exist?

Reply:
as of now yes. Does it make sense to create the directory if it doesn't exist?

Reply:
I think creating the directory is the way to go. The alternative is failing if the directory doesn't exist. That is less fat finger prone, but can also be annoying.